zapcore/syslog_encoder.go

```diff
@@ -0,0 +1,274 @@
+	return enc.je.AppendReflected(val)
+}
+
+func (enc *syslogEncoder) AppendString(val string) {
+	enc.je.AppendString(val)
+}
+
+func (enc *syslogEncoder) AppendTime(val time.Time) {
+	enc.je.AppendTime(val)
+}
+
+func (enc *syslogEncoder) AppendUint64(val uint64) {
+	enc.je.AppendUint64(val)
+}
+
+func (enc *syslogEncoder) AddComplex64(k string, v complex64) { enc.AddComplex128(k, complex128(v)) }
+func (enc *syslogEncoder) AddFloat32(k string, v float32)     { enc.AddFloat64(k, float64(v)) }
+func (enc *syslogEncoder) AddInt(k string, v int)             { enc.AddInt64(k, int64(v)) }
+func (enc *syslogEncoder) AddInt32(k string, v int32)         { enc.AddInt64(k, int64(v)) }
+func (enc *syslogEncoder) AddInt16(k string, v int16)         { enc.AddInt64(k, int64(v)) }
+func (enc *syslogEncoder) AddInt8(k string, v int8)           { enc.AddInt64(k, int64(v)) }
+func (enc *syslogEncoder) AddUint(k string, v uint)           { enc.AddUint64(k, uint64(v)) }
+func (enc *syslogEncoder) AddUint32(k string, v uint32)       { enc.AddUint64(k, uint64(v)) }
+func (enc *syslogEncoder) AddUint16(k string, v uint16)       { enc.AddUint64(k, uint64(v)) }
+func (enc *syslogEncoder) AddUint8(k string, v uint8)         { enc.AddUint64(k, uint64(v)) }
+func (enc *syslogEncoder) AddUintptr(k string, v uintptr)     { enc.AddUint64(k, uint64(v)) }
+func (enc *syslogEncoder) AppendComplex64(v complex64)        { enc.AppendComplex128(complex128(v)) }
+func (enc *syslogEncoder) AppendFloat64(v float64)            { enc.je.appendFloat(v, 64) }
+func (enc *syslogEncoder) AppendFloat32(v float32)            { enc.je.appendFloat(float64(v), 32) }
+func (enc *syslogEncoder) AppendInt(v int)                    { enc.AppendInt64(int64(v)) }
+func (enc *syslogEncoder) AppendInt32(v int32)                { enc.AppendInt64(int64(v)) }
+func (enc *syslogEncoder) AppendInt16(v int16)                { enc.AppendInt64(int64(v)) }
+func (enc *syslogEncoder) AppendInt8(v int8)                  { enc.AppendInt64(int64(v)) }
+func (enc *syslogEncoder) AppendUint(v uint)                  { enc.AppendUint64(uint64(v)) }
+func (enc *syslogEncoder) AppendUint32(v uint32)              { enc.AppendUint64(uint64(v)) }
+func (enc *syslogEncoder) AppendUint16(v uint16)              { enc.AppendUint64(uint64(v)) }
+func (enc *syslogEncoder) AppendUint8(v uint8)                { enc.AppendUint64(uint64(v)) }
+func (enc *syslogEncoder) AppendUintptr(v uintptr)            { enc.AppendUint64(uint64(v)) }
+
+func (enc *syslogEncoder) Clone() Encoder {
+	return enc.clone()
+}
+
+func (enc *syslogEncoder) clone() *syslogEncoder {
+	clone := getSyslogEncoder()
+	clone.facility = enc.facility
+	clone.je = enc.je.clone()
+	clone.pid = enc.pid
+	clone.hostname = enc.hostname
+	clone.tag = enc.tag
+	return clone
+}
+
+func (enc *syslogEncoder) EncodeEntry(ent Entry, fields []Field) (*buffer.Buffer, error) {
+	var p syslog.Priority
+	switch ent.Level {
```

Comment:
I'd mildly prefer to see this switch embodied in something like `func levelToSyslogPriority(Level) syslog.Priority`